basyx/aas/adapter/http.py

```diff
@@ -601,14 +610,23 @@ def handle_request(self, request: Request):
     def get_aas_all(self, request: Request, url_args: Dict, **_kwargs) -> Response:
         response_t = get_response_type(request)
         aas: Iterator[model.AssetAdministrationShell] = self._get_all_obj_of_type(model.AssetAdministrationShell)
+
+        # Filter by 'idShort' if the parameter is provided in the request
         id_short = request.args.get("idShort")
         if id_short is not None:
             aas = filter(lambda shell: shell.id_short == id_short, aas)
+
+        # Filtering by base64url encoded SpecificAssetIds if provided
         asset_ids = request.args.get("assetIds")
         if asset_ids is not None:
+            # Use the new method to decode and instantiate SpecificAssetIds
             spec_asset_ids = HTTPApiDecoder.base64urljson_list(asset_ids, model.SpecificAssetId, False, False)
-            # TODO: it's currently unclear how to filter with these SpecificAssetIds
-        return response_t(list(aas))
+            # Filter the AAS based on these SpecificAssetIds
+            aas = filter(lambda shell: all(
+                spec_id in [id_.identifier for id_ in shell.specific_asset_ids] for spec_id in spec_asset_ids), aas)
```

Comment:
`SpecificAssetId` doesn't have an `identifier` attribute, so this doesn't work. Furthermore, `AssetAdministrationShell` doesn't have a `specific_asset_ids` attribute. The `SpecificAssetIds` are located in the `AssetInformation`.